p2pserver/message/utils/msg_workerpool.go

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright (C) 2018 The ontology Authors
+ * This file is part of The ontology library.
+ *
+ * The ontology is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The ontology is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with The ontology.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package utils
+
+import (
+	"runtime"
+	"sync"
+	"time"
+
+	"github.com/ontio/ontology-eventbus/actor"
+	"github.com/ontio/ontology/common/log"
+	"github.com/ontio/ontology/p2pserver/message/types"
+	"github.com/ontio/ontology/p2pserver/net/protocol"
+)
+
+type msgJobChanList *[]*msgJobChan
```

Comment:
???

Reply:
The is a slice pointer, the slice element is a pointer.  A  better way for this to provide some comon libs for these basic data struct in the whole platform .

Reply:
I think slice is enough, why need a slice pointer?

Reply:
From the point of function,  slice is enough. The purpose of slice pointer is to reduce the copy and to be convenient for operation.